src/ctap/apdu.rs

```diff
@@ -168,7 +174,10 @@ impl TryFrom<&[u8]> for APDU {
                 return Err(ApduStatusCode::SW_WRONG_LENGTH);
             }
 
-            let extended_apdu_le_len: usize = payload.len() - extended_apdu_lc - 3;
+            let extended_apdu_le_len: usize = payload
+                .len()
+                .checked_sub(extended_apdu_lc + 3)
+                .unwrap_or(0xff);
```

Comment:
You are implicitly trying to reuse the error code handling in the `if` below? Should work, but explicit is usually better than implicit. That's why I proposed `ok_or(ERROR_CODE)?`.

Reply:
done

Reply:
Opened a new comment for a new possible issue.